packages/shortest/src/core/runner/test-run-repository.ts

```diff
@@ -0,0 +1,411 @@
+import * as fs from "fs/promises";
+import path from "path";
+import { CACHE_DIR_PATH } from "@/cache";
+import { TestCase } from "@/core/runner/test-case";
+import { TestRun } from "@/core/runner/test-run";
+import { getLogger, Log } from "@/log";
+import { CacheEntry } from "@/types/cache";
+import { getErrorDetails } from "@/utils/errors";
+
+// Shared process handlers registration
+let handlersRegistered = false;
+const registerSharedProcessHandlers = (log: Log) => {
+  if (handlersRegistered) return TestRunRepository.activeRepositories;
+
+  const activeRepositories = new Set<TestRunRepository>();
+
+  const cleanUpAndExit = async () => {
+    await Promise.all(
+      [...activeRepositories].map((repo) => repo.releaseLock()),
+    );
+    process.exit();
+  };
+
+  process.on("exit", cleanUpAndExit);
+  process.on("SIGINT", cleanUpAndExit);
+  process.on("SIGTERM", cleanUpAndExit);
+  process.on("uncaughtException", async (error) => {
+    log.error("Uncaught exception", getErrorDetails(error));
+    await cleanUpAndExit();
+  });
+
+  handlersRegistered = true;
+  return activeRepositories;
+};
+
+export class TestRunRepository {
+  public static VERSION = 2;
```

Comment:
Increased from `1` previously, and using numbers now. Generally, test runs with previous versions are automatically discarted, their metadata is different.